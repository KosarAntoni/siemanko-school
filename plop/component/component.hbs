import React from 'react';
import clsx from 'clsx';
{{#if styles}}

import styles from './{{properCase name}}.styles.module.scss';
{{/if}}
{{#if models}}

import { {{properCase name}}Props } from './{{properCase name}}.models';
{{/if}}

const {{properCase name}} = ({ className }: {{#if models}}{{properCase name}}Props{{/if}}): JSX.Element => (
  <div className={clsx(styles.{{camelCase name}}, className)} data-testid="{{dashCase name}}-item">
    {' '}
  </div>
);

export default {{properCase name}};
