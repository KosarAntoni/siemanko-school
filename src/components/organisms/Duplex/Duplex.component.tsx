'use client';
import React, { useRef } from 'react';
import { documentToReactComponents } from '@contentful/rich-text-react-renderer';
import clsx from 'clsx';
import Image from 'next/image';

import styles from './Duplex.styles.module.scss';

import { DuplexProps } from './Duplex.models';

import Button from '@/components/atoms/Button/Button.component';
import useElementPosition from '@/hooks/useElementPosition';

export const getOffset = (position: number, sectionHeight?: number): number => {
  if (!sectionHeight) return 0;

  return Math.round((position / (sectionHeight / 50)) * -1);
};

const Duplex = ({
  className,
  image,
  headline,
  button,
  description,
  imagePosition = 'Left',
}: DuplexProps): JSX.Element => {
  const sectionsRef = useRef<HTMLElement>(null);
  const position = useElementPosition(sectionsRef);

  const sectionHeight = sectionsRef?.current?.offsetHeight;

  return (
    <section
      ref={sectionsRef}
      className={clsx(styles.duplex, imagePosition === 'Right' && styles.duplexImageOnRight, className)}
      data-testid="duplex-item"
    >
      <Image
        src={image.url}
        alt={image.description}
        width={560}
        height={700}
        style={{ ['--translate-y' as string]: `${getOffset(position, sectionHeight)}px` }}
      />
      <div className={styles.duplexContent}>
        <h2>{headline}</h2>
        {description?.json && documentToReactComponents(description.json)}
        {button?.text && <Button {...button} />}
      </div>
    </section>
  );
};

export default Duplex;
