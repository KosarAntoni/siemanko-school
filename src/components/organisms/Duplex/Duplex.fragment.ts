export const duplexFragment = `fragment DuplexFragment on ComponentDuplex {
  sys {
    id
  }
  image {
    description
    url
  }
  imagePosition
  headline
  description {
    json
  }
  button {
    ...ButtonFragment
  }
}`;
