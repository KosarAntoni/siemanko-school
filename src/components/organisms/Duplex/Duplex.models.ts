import { Document } from '@contentful/rich-text-types';

import { ButtonProps } from '@/components/atoms/Button';

export interface DuplexProps {
  headline: string;
  image: {
    url: string;
    description: string;
  };
  className?: string;
  description?: {
    json: Document;
  };
  button?: ButtonProps;
}
