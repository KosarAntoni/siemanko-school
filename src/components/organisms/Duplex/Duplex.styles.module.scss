@import 'mixins';

.duplex {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 14rem;
  padding: 0 var(--side-padding) 3rem;
  background-color: var(--grey);

  @include media(sm) {
    flex-direction: row;
    padding: 0 4rem 10rem;
  }

  &ImageOnRight {
    img {
      order: 2;

      @include media(sm) {
        width: 50%;
        height: 500px;
      }

      @include media(lg) {
        width: 40%;
        height: 700px;
        margin-right: 8%;
      }
    }
  }

  img {
    width: 100%;
    height: 350px;
    margin-top: -8rem;
    object-fit: cover;
    border-radius: 0.5rem;
    transform: translateY(var(--translate-y));
    transition: transform 0.1s;

    @include media(sm) {
      width: 50%;
      height: 500px;
    }

    @include media(lg) {
      width: 40%;
      height: 700px;
      margin-left: 8%;
    }
  }

  &Content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    padding-top: 4rem;

    @include media(sm) {
      margin: 10% 4% 4%;
    }

    @include media(lg) {
      width: 40%;
      margin: 10% 8% 4%;
    }

    h2 {
      position: relative;
      padding: 3rem 0 2rem;
      font-size: 3rem;

      @include media(lg) {
        font-size: 4.8rem;
      }

      &::before {
        position: absolute;
        top: 0;
        display: block;
        width: 8rem;
        height: 0.5rem;
        background-color: var(--orange);
        content: '';
      }
    }

    p {
      margin-bottom: 4rem;
      font-size: 1.6rem;
      line-height: 1.5;

      @include media(lg) {
        font-size: 2rem;
      }
    }
  }
}
