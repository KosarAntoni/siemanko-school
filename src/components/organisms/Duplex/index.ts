export { default } from './Duplex.component';
export * from './Duplex.models';
