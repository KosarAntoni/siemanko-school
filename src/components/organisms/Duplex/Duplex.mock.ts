import { DuplexProps } from './Duplex.models';

export const mock = {
  headline: 'Kilka słów o nas',
  description: {
    json: {
      nodeType: 'document',
      data: {},
      content: [
        {
          nodeType: 'paragraph',
          data: {},
          content: [
            {
              nodeType: 'text',
              value:
                'Wszystko zaczęło się od pasji, która dziś dokonuje rewolucji w życiu naszym i naszych klientów. Wiedzieliśmy, że dokonamy wiele. Nie przypuszczaliśmy jednak, że prawdziwa pasja i systematyczna praca, tak szybko zaprowadzi nas na sam szczyt w ciągu zaledwie 5 lat!',
              marks: [],
              data: {},
            },
          ],
        },
      ],
    },
  },
  button: {
    text: 'CZYTAJ WIĘCEJ',
    url: null,
    reference: null,
    style: 'Outline',
    color: 'Orange',
    size: 'Big',
  },
  image: {
    description: 'Cała ekipa instruktorów Siemanko',
    url: 'https://images.ctfassets.net/39emsr6r0xu2/6VZJiGrlzCmDjIZdXNJkYe/eaf38e615f003e9f5f125031ee143296/zaloga_siemanko_01.jpg',
  },
} as DuplexProps;
