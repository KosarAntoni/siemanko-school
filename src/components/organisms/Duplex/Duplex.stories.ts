import type { Meta, StoryObj } from '@storybook/react';

import Duplex from '.';
import { mock } from './Duplex.mock';

const meta: Meta<typeof Duplex> = {
  title: 'Organisms/Duplex',
  component: Duplex,
  tags: ['autodocs'],
  args: { ...mock },
};

export default meta;
type Story = StoryObj<typeof Duplex>;

export const Default: Story = {
  args: {},
};
