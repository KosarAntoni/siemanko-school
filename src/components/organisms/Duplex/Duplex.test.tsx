import React from 'react';
import { render, screen } from '@testing-library/react';

import Duplex from '.';
import { getOffset } from './Duplex.component';
import { mock } from './Duplex.mock';

describe('Duplex component renders correctly', () => {
  it('renders Duplex', () => {
    render(<Duplex {...mock} />);

    const duplexNode = screen.getByTestId('duplex-item');

    expect(duplexNode).toBeInTheDocument();

    expect(duplexNode).toMatchSnapshot();

    const imageNode = screen.getByAltText(mock.image.description);
    expect(imageNode).toBeInTheDocument();
    expect(imageNode).toHaveAttribute('style', 'color: transparent; --translate-y: 0px;');
  });

  it('renders Duplex with image on right side', () => {
    render(<Duplex {...mock} imagePosition="Right" />);

    const duplexNode = screen.getByTestId('duplex-item');

    expect(duplexNode).toBeInTheDocument();

    expect(duplexNode).toMatchSnapshot();

    expect(duplexNode).toHaveClass('duplexImageOnRight');
  });

  it('getOffset works correctly', () => {
    expect(getOffset(36, 573)).toBe(-3);
    expect(getOffset(-124, 720)).toBe(9);
    expect(getOffset(-124)).toBe(0);
  });
});
