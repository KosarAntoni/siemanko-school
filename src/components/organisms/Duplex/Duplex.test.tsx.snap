// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duplex component renders correctly renders Duplex 1`] = `
<section
  class="duplex"
  data-testid="duplex-item"
>
  <img
    alt="Cała ekipa instruktorów Siemanko"
    data-nimg="1"
    decoding="async"
    height="700"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=1200&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=1200&q=75 2x"
    style="color: transparent; --translate-y: 0px;"
    width="560"
  />
  <div
    class="duplexContent"
  >
    <h2>
      Kilka słów o nas
    </h2>
    <p>
      Wszystko zaczęło się od pasji, która dziś dokonuje rewolucji w życiu naszym i naszych klientów. Wiedzieliśmy, że dokonamy wiele. Nie przypuszczaliśmy jednak, że prawdziwa pasja i systematyczna praca, tak szybko zaprowadzi nas na sam szczyt w ciągu zaledwie 5 lat!
    </p>
    <div
      class="button buttonOrange buttonOutline buttonBig"
      data-testid="button-item"
    >
      CZYTAJ WIĘCEJ
    </div>
  </div>
</section>
`;

exports[`Duplex component renders correctly renders Duplex with image on right side 1`] = `
<section
  class="duplex duplexImageOnRight"
  data-testid="duplex-item"
>
  <img
    alt="Cała ekipa instruktorów Siemanko"
    data-nimg="1"
    decoding="async"
    height="700"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=1200&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F39emsr6r0xu2%2F6VZJiGrlzCmDjIZdXNJkYe%2Feaf38e615f003e9f5f125031ee143296%2Fzaloga_siemanko_01.jpg&w=1200&q=75 2x"
    style="color: transparent; --translate-y: 0px;"
    width="560"
  />
  <div
    class="duplexContent"
  >
    <h2>
      Kilka słów o nas
    </h2>
    <p>
      Wszystko zaczęło się od pasji, która dziś dokonuje rewolucji w życiu naszym i naszych klientów. Wiedzieliśmy, że dokonamy wiele. Nie przypuszczaliśmy jednak, że prawdziwa pasja i systematyczna praca, tak szybko zaprowadzi nas na sam szczyt w ciągu zaledwie 5 lat!
    </p>
    <div
      class="button buttonOrange buttonOutline buttonBig"
      data-testid="button-item"
    >
      CZYTAJ WIĘCEJ
    </div>
  </div>
</section>
`;
